lib/explorer/backend/lazy_series.ex

```diff
@@ -635,6 +661,51 @@ defmodule Explorer.Backend.LazySeries do
   defp resolve_numeric_dtype(:window_mean, _items), do: :float
   defp resolve_numeric_dtype(_op, items), do: resolve_numeric_dtype(items)
 
+  defp resolve_numeric_temporal_dtype(op, %Series{dtype: ldt} = left, %Series{dtype: rdt} = right) do
+    case {op, ldt, rdt} do
+      {:add, {:datetime, ltu}, {:duration, rtu}} -> {:datetime, highest_precision(ltu, rtu)}
+      {:add, {:duration, ltu}, {:datetime, rtu}} -> {:datetime, highest_precision(ltu, rtu)}
+      {:add, {:duration, ltu}, {:duration, rtu}} -> {:duration, highest_precision(ltu, rtu)}
+      {:subtract, {:datetime, ltu}, {:datetime, rtu}} -> {:duration, highest_precision(ltu, rtu)}
+      {:subtract, {:datetime, ltu}, {:duration, rtu}} -> {:datetime, highest_precision(ltu, rtu)}
+      {:subtract, {:duration, ltu}, {:duration, rtu}} -> {:duration, highest_precision(ltu, rtu)}
+      {:multiply, :integer, {:duration, tu}} -> {:duration, tu}
+      {:multiply, {:duration, tu}, :integer} -> {:duration, tu}
+      {:divide, {:duration, tu}, :integer} -> {:duration, tu}
+      {:divide, _, _} -> :float
+      _ -> resolve_numeric_dtype([left, right])
+    end
```

Comment:
:+1: for more tests. I am just not sure if property based testing helps because it will be mostly hardcoding the rules above. :(

Reply:
> So many possibilities :O

Yeah it's not my most elegant code 😅. Especially because it's duplicated in a few places. _Definitely_ open to any thoughts on simplifying, though I suspect all versions will be similarly gnarly.

> This may be too much, but if we can add some property tests, it can be easier to guarantee some part of the behaviour.

I think I agree with @josevalim: we'd probably just be duplicating the same logic, meaning the properties don't add anything. However, I can envision properties like the following:

* `add`/`subtract` and `multiply`/`divide` are inverse operations (modulo infinity), so things like `x |> add(y) |> subtract(y)` should either `== x` or type error

Proposal: I'll focus on exhaustively testing the known cases up front then come back and see if there are any properties which make sense. Thoughts?

Reply:
Perfect.

Reply:
Partially addressed by: [7e7b961](https://github.com/elixir-explorer/explorer/pull/683/commits/7e7b961f834ea41c40fec1b433b70312b8c0339c)

I've got a note to revisit the property tests in my to-dos.